lib/identity_cache/query_api.rb

```diff
@@ -401,13 +401,15 @@ def fetch_recursively_cached_association(ivar_name, association_name) # :nodoc:
       if IdentityCache.should_use_cache?
```

Comment:
move `assoc = ` to this line and 

```ruby
        assoc = assoc.to_ary if assoc.respond_to?(:to_ary) && !IdentityCache.fetch_returns_relation
        assoc
```

after this condition.  That way the else block can be changed back to `send(association_name.to_sym)` and the conditional array coercion only needs to happen in one place in this method